### DISCUSSION TITLE
Disallowed range operator after route set name in `<filter>`

### DISCUSSION THREAD

**Discussion by SichangHe** [SichangHe/internet_route_verification#51] [General]
> * [ ]  Allow range operator after any route set in filter.
> 
> > A route set name can also be followed by one of the operators '^-',
> > '^+', example, { 5.0.0.0/8, 6.0.0.0/8 }^+ equals { 5.0.0.0/8^+,
> > 6.0.0.0/8^+ }, and AS1^- equals all the exclusive more specifics of
> > routes originated by AS1.

I just doublechecked this rule. We do comply with them.

The large number of warnings I see are "route set names" followed by the other operators—`^n` and `^m-n`. Base on this rule in the RFC, those are not allowed, but apparently, they are used everywhere.

_Originally posted by @SichangHe in https://github.com/SichangHe/internet_route_verification/issues/41#issuecomment-1703671305_

**Comment by SichangHe** [comment]
I counted 4724 of them by parsing the log.

[disallowed_range_operator_in_filter.txt](https://github.com/SichangHe/internet_route_verification/files/12504928/disallowed_range_operator_in_filter.txt)

**Comment by SichangHe** [comment]
I could not find any rules that allow this in RFC 2622, 2650, or 4012.

But, since they are frequently used, I will allow them.

**Comment by SichangHe** [comment]
I have allowed *any* range operators following route set names. The number of lines of logs indeed reduced by about 5000.

**Comment by cunha** [comment]
This may be an oversight in the writing of the RFC. One thing we could do is track down these ambiguities or missing information, and we could propose them as errata later on.

**Reply by SichangHe** [reply]
I propose that we go over every issue and label them as either `noted` or `wontnote` after adding them (or figuring out not to add) in the writing.

**Reply by cunha** [reply]
For the paper, we could also `note` everything, and then choose what to actually include in the PDF before the submission. (In other words, have all the content available, but choose exactly what to include as we go.)
